files/scripts/supervisor-proc-exit-listener

```diff
@@ -35,12 +52,21 @@ def main():
             processname = payload_headers['processname']
             groupname = payload_headers['groupname']
 
-            # If a critical process exited unexpectedly, terminate supervisor
-            if expected == 0 and processname in critical_processes or groupname in critical_processes:
+            config_db = swsssdk.ConfigDBConnector()
+            config_db.connect()
+            docker_config = config_db.get_table(CONTAINER_FEATURE_TABLE_NAME)
+            if docker_config and docker_config.has_key(container_name):
+                restart_feature = docker_config[container_name].get('auto_restart')
+                if not restart_feature:
+                    syslog.syslog(syslog.LOG_ERROR, "Unable to determine auto-restart feature status for container '{}'. Exiting".format(container_name))
+                    sys.exit(1)
```

Comment:
Change this exit code to 2 to differentiate from the other abnormal exit.

Reply:
Updated.